.github/workflows/main.yml

```diff
@@ -100,35 +97,33 @@ jobs:
 
     - name: Generate the revision file
       run: |
-        texlua build.lua revisionfile
+        l3build revisionfile
```

Comment:
CI fails at here. Local tests show `l3build revisionfile` returns exit code 0 but `texlua build.lua revisionfile` works fine.

Is a local function defined in `build.lua` automatically capable of a `<target>` in `l3build <target>`? Sorry I have very limited knowledge/experience about both Lua and l3build.

Reply:
> Is a local function defined in `build.lua` automatically capable of a `<target>` in `l3build <target>`?

Seems `revisionfile` is already registered to `target_list`, see https://github.com/pgf-tikz/pgf/pull/1094/files#diff-9e218184846f8e9c379402b4d9ddea5abac901ca71d87dcaf61713467c19db42R96-R101

Reply:
I'd not added a return value, and it seems the YAML wants an explicit zero rather than 'N/A'.

Reply:
> Local tests show `l3build revisionfile` returns exit code 0

typo: The exit code was 1, not 0.

Is it possible that it's `l3build` that sets the exit code 1? I notice [`main(target, names)`](https://github.com/latex3/l3build/blob/main/l3build-stdmain.lua#L154) in [`stdmain.lua`](https://github.com/latex3/l3build/blob/main/l3build-stdmain.lua) has this logic:
```lua
  if errorlevel ~= 0 then
    exit(1)
  else
    exit(0)
```
and since the default return value of a lua function is `nil` ...